Shell/NewCNF.cpp

```diff
@@ -993,7 +993,7 @@ Term* NewCNF::createSkolemTerm(unsigned var, VarSet* free, Formula *reuse_formul
   ASS(domainSorts.isEmpty());
   ASS(fnArgs.isEmpty());
 
-  NameReuse *name_reuse = env.options->skolemReuse()
+  NameReuse *name_reuse = (env.options->skolemReuse() && reuse_formula)
```

Comment:
Isn't `reuse_formula` only ever `nullptr` if `!env.options->skolemReuse()`? Think this might be a copy/paste error from the other fix.

Reply:
`(reuse_formula == nullptr)` with `env.options->skolemReuse()` is the bailout situation with `(VList::length(g->vars()) > 5)`